src/struct.rb

```diff
@@ -73,10 +73,31 @@ def self.new(*attrs)
         define_method :[] do |arg|
           case arg
           when Integer
-            attribute = attrs.fetch(arg)
-            send(attribute)
+            arg = attrs.fetch(arg)
+          when String, Symbol
+            unless attrs.include?(arg.to_sym)
+              raise NameError, "no member '#{arg}' in struct"
+            end
+          end
+          send(arg)
+        end
+
+        define_method :dig do |*args|
+          if args.empty?
+            raise ArgumentError, 'wrong number of arguments (given 0, expected 1+)'
+          end
+          arg = args.shift
```

Comment:
This looks very convoluted, but it seems like `define_method :dig do |arg, *args|` does not validate the size of the arguments, which means `arg` would be set to `nil` instead of raising an `ArgumentError`